### PR TITLE
Makes it so you actually SPIN YOUR GUN while gun-spinning

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
My Magnum Opus, it is done after all this time
spin2 gof

## Why It's Good For The Game
This is badass

## Changelog
:cl: itseasytosee
qol: Your gun will now actually spin around in your hand while gun-spinning
/:cl:

### CODE REVIEW

**Comment by SmArtKar** [comment]
can you make it like 3 times faster but make it spin 3 times?

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -440,6 +440,7 @@
 
 /obj/item/gun/ballistic/attack_self(mob/living/user)
 	if(HAS_TRAIT(user, TRAIT_GUNFLIP))
+		src.SpinAnimation(7,1)
```
Comment (Fikou) [inline code/modules/projectiles/guns/ballistic.dm:443]: dont do src.proc

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -440,6 +440,7 @@
...
 		if(flip_cooldown <= world.time)
 			if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(40))
 				to_chat(user, span_userdanger("While trying to flip the [src] you pull the trigger and accidently shoot yourself!"))
```
Comment (Fikou) [inline code/modules/projectiles/guns/ballistic.dm:446]: might be out of scope but "the [src]" should either be "\the [src]" or "[src]"

File: code/modules/projectiles/guns/ballistic.dm (hunk 1 above)
Comment (itseasytosee) [reply code/modules/projectiles/guns/ballistic.dm:443]: I'm not aware of the policy on this, what should I do instead.

Comment (MrDoomBringer) [reply code/modules/projectiles/guns/ballistic.dm:443]:
```suggestion
		SpinAnimation(7,1)
```

Comment (MrDoomBringer) [reply code/modules/projectiles/guns/ballistic.dm:443]: like so

Comment (itseasytosee) [reply code/modules/projectiles/guns/ballistic.dm:443]: Thanks

**Comment by Mothblocks** [comment]
if you changed the animation show the new one

**Comment by itseasytosee** [comment]
Here it is
spin2 gof

And here's what it looks like when you spam Z
spin3

**Comment by Mothblocks** [comment]
still a bit slow  imo but no point bikeshedding on it

File: code/modules/projectiles/guns/ballistic.dm [hunk 3]
```diff
@@ -440,9 +440,10 @@
 
 /obj/item/gun/ballistic/attack_self(mob/living/user)
 	if(HAS_TRAIT(user, TRAIT_GUNFLIP))
+		SpinAnimation(4,2)
 		if(flip_cooldown <= world.time)
 			if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(40))
-				to_chat(user, span_userdanger("While trying to flip the [src] you pull the trigger and accidently shoot yourself!"))
+				to_chat(user, span_userdanger("While trying to flip [src] you pull the trigger and accidently shoot yourself!"))
```
Comment (maxymax13) [inline code/modules/projectiles/guns/ballistic.dm:446]: you left in "accidently" >:(

Comment (itseasytosee) [reply code/modules/projectiles/guns/ballistic.dm:446]: 😿

**Pull request merged**